python/cudf/cudf/tests/test_rolling.py

```diff
@@ -124,6 +131,78 @@ def test_rolling_with_offset(agg):
     )
 
 
+def generate_large_dataframe_for_var(size, window_size, nulls_prob, seed):
```

Comment:
Can you add some comments describing the contents of the DataFrame that you're generating? It's not clear if this choice of data is random or if there's a specific reason for it.

Reply:
I wonder if the parameterization you're using is related to the test failure right now?

Reply:
> It's not clear if this choice of data is random or if there's a specific reason for it.

I mainly wrote this to replace original use of `dataset_generator`, which generates values far too large for computing anything meaningful for var (which has a sqaured sum).

Reply:
> I wonder if the parameterization you're using is related to the test failure right now?

I think failed test case is with `test_rolling_series_basic` with data setup: `[nan, nan, 4, 9, 9, nan]`. The difference is with the window `[9, 9, nan]`. In pandas the variance is non zero because of [floating point artifact](https://pandas.pydata.org/pandas-docs/stable/whatsnew/v1.3.0.html#removed-artificial-truncation-in-rolling-variance-and-standard-deviation) (compensation term of Kahan summation). In libcudf the result is just 0.

I couldn't capture this error locally because the nans are randomly assigned in this test case. It happens that the uniform window is left intact when running in CI. I should probably fix the seeds for these tests to avoid randomly breaking CIs in the future.

Reply:
Could you just add a parameter to the int/float generation in `dataset_generator` that is used in place of iinfo.min/max if provided?

Reply:
Floating point error is a fact of life, I think the correct solution should just be to pass an appropriate tolerance to `assert_eq`. That will get forwarded to the numpy/pandas testing tool that we're calling under the hood. Since the problem is a comparison to 0, you'll need to set `atol`, not `rtol`.

Reply:
> you'll need to set `atol`

But the following example shows that the error could vary based on the value of preceding window.

```python
>>> s = pd.Series([7, 5, 5, 5])
>>> s.rolling(3).var()
0             NaN
1             NaN
2    1.333333e+00
3    4.440892e-16
dtype: float64

>>> s = pd.Series([1e8, 5, 5, 5])
>>> s.rolling(3).var()
0             NaN
1             NaN
2    3.333333e+15
3   -5.000000e-01
dtype: float64
```

Reply:
> Could you just add a parameter to the int/float generation in dataset_generator that is used in place of iinfo.min/max if provided?

That's definitely an option too, which I believe at one point @galipremsagar and I discussed about. Prem are you aware if the number limits for the dataset generator is being implemented? Otherwise I'm happy to submit a PR.

Reply:
Yes, will open a PR shortly today

Reply:
@isVoid Here is the PR: https://github.com/rapidsai/cudf/pull/9211, could you review and approve it if that's fitting your needs here?

Reply:
I see, the running error counter is of magnitude scaled to the largest previously observed entry. That's unfortunate. I would still expect that you could use something like `atol=max(arr) * EPSILON` to capture the maximal deviation accurately more accurately, but that's a more work and not something that's really necessary since it's ultimately to correct for a pandas issue. Feel free to resolve this however you see fit.